src/Mobs/Monster.cpp

```diff
@@ -1294,3 +1401,60 @@ cMonster::eFamily cMonster::GetMobFamily(void) const
 {
 	return FamilyFromType(m_MobType);
 }
+
+
+
+
+
+void cMonster::LeashTo(cEntity * a_Entity, bool a_ShouldBroadcast)
+{
+	m_LeashedTo = a_Entity;
```

Comment:
Oh wait, is this the directly exported Lua API? Plugins are bound to call this while already leashed, because we plugin authors are stupid and too lazy to read docs. So make the function behave well in such a case - ideally re-leash the mob to the new entity, or if not practically possible, don't change the leash-ness.